docs/keymanager-developer-guide.md

```diff
@@ -0,0 +1,103 @@
+
+2. Unzip Apache Maven and move  `settings.xml` to "conf" folder `<apache maven unzip path>\conf`.
+
+3. Install Eclipse, open the `lombok.jar` file  and then click `Install/Update`.
+<img src="_images/lombok-configuration.png" width="750" height="450">
+
+4. Check the Eclipse installation folder to see if the `lombok.jar` is added.
+
+5. Configure the JDK (Standard VM) with your Eclipse by traversing through `Preferences → Java → Installed JREs`.
+<img src="_images/installed-jre.png" width="750" height="450">
+
+### Source code setup
+
+For the code setup, clone the repository and follow the guidelines mentioned in the [Code Contributions](https://docs.mosip.io/1.2.0/community/code-contributions).
+
+### Importing and building
+
+1. Open the project folder where `pom.xml` is present.
+2. Open the command prompt from the same folder.
+3. Run the command `mvn clean install -Dgpg.skip=true -DskipTests=true` to build the project .
+4. After building, open Eclipse and select `Import Projects → Maven → Existing Maven Projects → Next → Browse to project directory → Finish`.
+5. After successful importing of project, update the project by right-click on `Project → Maven → Update Project`.
+<img src="_images/import-project.png" width="750" height="450">
+
+## Environment setup
+
+1. Download [Auth adapter](https://oss.sonatype.org/#nexus-search;gav~~kernel-auth-adapter~1.2.0-SNAPSHOT~~) and add to project ```Libraries → Classpath → Add External JARs → Select Downloaded JAR → Add → Apply and Close```.
+<img src="_images/add-external-library.png" width="750" height="450">
+
+2. Clone [mosip-config repository](https://github.com/mosip/mosip-config).
+
+3. Refer [Keymanager-DB-deploy](https://github.com/mosip/keymanager/blob/develop/db_scripts/README.md) to deploy local DB.
+
+4. Keymanager uses two property files, `kernel-default` and `application-default`, configure them accordly.For instance
+* Keymanager needs a Keystore to store keys. Supported Keystore types: PKCS11, PKCS12, Offline, JCE. 
+
+    ```
+    # For PKCS11 provide Path of config file.
+    # For PKCS12 keystore type provide the p12/pfx file path. P12 file will be created internally so provide only file path & file name.
+    # For Offline & JCE property can be left blank, specified value will be ignored.
+    mosip.kernel.keymanager.hsm.config-path=/config/softhsm-application.conf
+    # Passkey of keystore for PKCS11, PKCS12
+    # For Offline & JCE proer can be left blank. JCE password use other JCE specific properties.
+    mosip.kernel.keymanager.hsm.keystore-pass={cipher}2d6aa328be521b2be6f33f476f7df2ea39c7ae1a3e2146ec169c5fac3225da3f
+    ```
+* Secrets can be encrypted using [config server](https://cloud.spring.io/spring-cloud-config/reference/html/#_encryption_and_decryption).
+* Update Url's in property files.(It can be either pointed to any remotely or locally deployed services)
```

Comment:
Update URL's in property files (can be either pointed to any remotely or locally deployed services).